l3_dump.py

```diff
@@ -79,13 +85,29 @@ def exec_binary(cmdargs:list):
         strings[offs] = s
 
 with open(sys.argv[1], 'rb') as file:
-    data = file.read(32)
+    # Unpack the 1st n-bytes as an L3_LOG{} struct to get a hold
+    # of the fbase-address stashed by the l3_init() call.
+    data = file.read(L3_LOG_HEADER_SZ)
     idx, loc, fibase, _, _ = struct.unpack('<iiQQQ', data)
     # print(f"{idx=} {fibase=:x}")
 
-    for _ in range(3):
-        row = file.read(32)
+    # pylint: disable-next=invalid-name
+    nentries = 0
+    # Keep reading chunks of log-entries from file ...
+    while True:
+        row = file.read(L3_ENTRY_SZ)
+        len_row = len(row)
+
+        # Deal with eof
+        if not row or len_row == 0 or len_row < L3_ENTRY_SZ:
```

Comment:
This check is added a bit defensively.

- `not row` is ok; the `file.read()` is doc'ed to return nothing upon eof.
- `len_row == 0` also seems reasonable to check for.
- Not so so sure about `len_row < L3_ENTRY_SZ`. I found this necessary for some tests that would, otherwise, run off into endless loops or print bad output upon reaching what-seems-to-be like an eof.

In any case, if we do hit the 3rd condition, we are looking at an incomplete log-entry, which will be difficult to parse and unpack (anyway).